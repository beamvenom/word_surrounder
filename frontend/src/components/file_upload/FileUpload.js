import styled from 'styled-components'

export const FileUploadContainer = styled.section`
  position: relative;
  vertical-align: bottom;
  border: 2px dotted lightgray;
  padding: 35px 20px;
  display: flex;
  flex-direction: column;
  align-items: center;
  background-color: white;
`

export const FormField = styled.input`
  font-size: 18px;
  display: block;
  width: 100%;
  border: none;
  text-transform: none;
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  opacity: 0;

  &:focus {
    outline: none;
  }
`

export const FileUploadText = styled.p`
  font-weight: bold;
  letter-spacing: 2.2px;
  margin-top: 0;
  text-align: center;
`

export const UploadFileBtn = styled.button`
  box-sizing: border-box;
  appearance: none;
  background-color: transparent;
  border: 2px solid #3498db;
  cursor: pointer;
  font-size: 1rem;
  line-height: ;
  padding: 1.1em 2.8em;
  text-align: center;
  text-transform: uppercase;
  font-weight: 700;
  border-radius: 6px;
  color: #3498db;
  position: relative;
  overflow: hidden;
  z-index: 1;
  transition: color 200ms ease-in-out;
  font-family: 'Open Sans', sans-serif;
  width: 45%;
  display: flex;
  align-items: center;
  padding-right: 0;
  justify-content: center;

  &:after {
    content: '';
    position: absolute;
    display: block;
    top: 0;
    left: 50%;
    transform: translateX(-50%);
    width: 0;
    height: 100%;
    background: #3498db;
    z-index: -1;
    transition: width 250ms ease-in-out;
  }

  i {
    font-size: 30px;
    margin-right: 5px;
    border-right: 2px solid;
    position: absolute;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
    width: 20%;
    display: flex;
    flex-direction: column;
    justify-content: center;
  }

  @media only screen and (max-width: 500px) {
    width: 70%;
  }

  @media only screen and (max-width: 350px) {
    width: 100%;
  }

  &:hover {
    color: #fff;
    outline: 0;
    background: transparent;

    &:after {
      width: 110%;
    }
  }

  &:focus {
    outline: 0;
    background: transparent;
  }

  &:disabled {
    opacity: 0.4;
    filter: grayscale(100%);
    pointer-events: none;
  }
`
