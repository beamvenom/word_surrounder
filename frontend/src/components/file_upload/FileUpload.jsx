import React from 'react'
import {
  FileUploadContainer,
  FormField,
  FileUploadText,
  UploadFileBtn,
} from './FileUpload.js'

const FileUpload = (props) => {
  const onFileChange = () => {
    var uploadedFile = document.querySelector('input[type=file]').files[0]
    var allowedFileFormats = ['txt', 'rtf', 'md', 'file']
    if (typeof uploadedFile !== 'undefined') {
      if (allowedFileFormats.includes(uploadedFile.name.split('.')[1])) {
        document.getElementById('uploadText').innerHTML = '...Processing...'
        props.setUploadedFile(uploadedFile)
      } else {
        document.getElementById('uploadText').innerHTML =
          'The supported file formats are .txt .rtf .file .md'
      }
    }
  }
  return (
    <>
      <FileUploadContainer>
        <FileUploadText type data-testid="uploadText" id="uploadText">
          Upload a text and the most occuring word in the text will be surronded by foo
          and bar!
        </FileUploadText>
        <UploadFileBtn>
          <i className="fas fa-file-upload" />
          <span data-testid = "fileFieldText"> Upload a text file </span>
          <FormField data-testid = "fileField" type="file" onChange={onFileChange} />
        </UploadFileBtn>
      </FileUploadContainer>
    </>
  )
}
export default FileUpload