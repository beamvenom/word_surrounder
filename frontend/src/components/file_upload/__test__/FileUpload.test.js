import React from 'react'
import { render, fireEvent , screen} from '@testing-library/react'
import FileUpload from '../FileUpload.jsx'
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom'

let setUploadedFile
beforeEach(() => {
    setUploadedFile = jest.fn()
    render(<FileUpload setUploadedFile = {setUploadedFile}></FileUpload>)
    
  });

test("Upload text renders correctly",() =>{
    expect(screen.getByTestId("uploadText")).toHaveTextContent("Upload a text and the most occuring word in the text will be surronded by foo and bar!")
})
test("File field text renders correctly",() =>{
    expect(screen.getByTestId("fileFieldText")).toHaveTextContent("Upload a text file")
})
test("File field accepts text file correctly",() =>{
    const file = new File(['hello'], 'text-file.txt', { type: 'text/plain' })
    fireEvent.change(screen.getByTestId("fileField"), {
        target: { files: [file]},
    })
    let text = document.querySelector('input[type=file]')
    expect(text.files[0].name).toBe('text-file.txt')
    expect(screen.getByTestId("uploadText")).toHaveTextContent('...Processing...')
    expect(setUploadedFile).toHaveBeenCalledTimes(1)
})
test("File field denies incorrect file format correctly",() =>{
    const file = new File(['(⌐□_□)'], 'png-file.png', { type: 'image/png' })
    fireEvent.change(screen.getByTestId("fileField"), {
        target: { files: [file]},
    })
    let image = document.querySelector('input[type=file]')
    expect(image.files[0].name).toBe('png-file.png')
    expect(screen.getByTestId("uploadText")).toHaveTextContent('The supported file formats are .txt .rtf .file .md')
    expect(setUploadedFile).toHaveBeenCalledTimes(0)
})

test("matches snapshot", () => {
    const tree = renderer.create(<FileUpload></FileUpload>)
    expect(tree).toMatchSnapshot()
})