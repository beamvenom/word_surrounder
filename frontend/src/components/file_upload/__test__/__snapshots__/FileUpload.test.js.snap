// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<section
  className="sc-bdfBQB isbfYM"
>
  <p
    className="sc-dlfnuX goCNQr"
    data-testid="uploadText"
    id="uploadText"
    type={true}
  >
    Upload a text and the most occuring word in the text will be surronded by foo and bar!
  </p>
  <button
    className="sc-hKgJUU AwSXH"
  >
    <i
      className="fas fa-file-upload"
    />
    <span
      data-testid="fileFieldText"
    >
       Upload a text file 
    </span>
    <input
      className="sc-gsTEea kuEFkz"
      data-testid="fileField"
      onChange={[Function]}
      type="file"
    />
  </button>
</section>
`;
